apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/[keyId]/settings/update-key-remaining.tsx

```diff
@@ -116,23 +93,20 @@ export const UpdateKeyRemaining: React.FC<Props> = ({ apiKey }) => {
   });
 
   async function onSubmit(values: z.infer<typeof formSchema>) {
-    if (values.refill?.interval !== "none" && !values.refill?.amount) {
+    if (!values.refill?.amount && values.refill?.refillDay) {
       form.setError("refill.amount", {
-        message: "Please enter the number of uses per interval",
+        message: "Please enter the number of uses per interval or remove the refill day",
       });
       return;
     }
-    if (values.refill.interval !== "none" && values.remaining === undefined) {
+    if (values.remaining === undefined) {
```

Comment:
Let's go with `values.remaining == null` or `values.remaining == undefined` to cover both cases.

See here to understand why this works if you are curious 😄 
https://tc39.es/ecma262/#sec-abstract-equality-comparison
https://developer.mozilla.org/en-US/docs/Glossary/Type_coercion

Reply:
Sounds Good